src/coreComponents/physicsSolvers/contact/SolidMechanicsAugmentedLagrangianContact.cpp

```diff
@@ -0,0 +1,1347 @@
+                                               dofManager,
+                                               localMatrix,
+                                               localRhs );
+
+  //ParallelMatrix parallel_matrix;
+  //parallel_matrix.create( localMatrix.toViewConst(), dofManager.numLocalDofs(), MPI_COMM_GEOSX );
+  //parallel_matrix.write("mech.mtx");
+  //abort();
+
+  // Loop for assembling contributes from interface elements (Aut*eps^-1*Atu and Aub*eps^-1*Abu)
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const & meshName,
+                                                                MeshLevel & mesh,
+                                                                arrayView1d< string const > const & )
+
+  {
+
+    NodeManager const & nodeManager = mesh.getNodeManager();
+    FaceManager const & faceManager = mesh.getFaceManager();
+
+    string const & dispDofKey = dofManager.getKey( solidMechanics::totalDisplacement::key() );
+    string const & bubbleDofKey = dofManager.getKey( solidMechanics::totalBubbleDisplacement::key() );
+
+    arrayView1d< globalIndex const > const dispDofNumber = nodeManager.getReference< globalIndex_array >( dispDofKey );
+    arrayView1d< globalIndex const > const bubbleDofNumber = faceManager.getReference< globalIndex_array >( bubbleDofKey );
+
+    string const & fractureRegionName = this->getUniqueFractureRegionName();
+
+    forFiniteElementOnFractureSubRegions( meshName, [&] ( string const & finiteElementName,
+                                                          arrayView1d< localIndex const > const & faceElementList )
+    {
+
+      finiteElement::FiniteElementBase & subRegionFE = *(m_faceTypeToFiniteElements[finiteElementName]);
+
+      solidMechanicsALMKernels::ALMFactory kernelFactory( dispDofNumber,
+                                                          bubbleDofNumber,
+                                                          dofManager.rankOffset(),
+                                                          localMatrix,
+                                                          localRhs,
+                                                          dt,
+                                                          faceElementList );
+
+      real64 maxTraction = finiteElement::
+                             interfaceBasedKernelApplication
+                           < parallelDevicePolicy< >,
+                             constitutive::NullModel >( mesh,
+                                                        fractureRegionName,
+                                                        faceElementList,
+                                                        subRegionFE,
+                                                        "",
+                                                        kernelFactory );
+
+      GEOS_UNUSED_VAR( maxTraction );
+
+    } );
+
+  } );
+
+  // Loop for assembling contributes of bubble elements (Abb, Abu, Aub)
```

Comment:
you may want to break this down into multiple functions.

Reply:
Is it convenient?